src/main/scala/com/linkedin/feathr/offline/config/FeathrConfigLoader.scala

```diff
@@ -254,14 +255,23 @@ private[offline] class AnchorLoader extends JsonDeserializer[FeatureAnchor] {
     // if it is UDF, no extra information other than the class name is required
     val anchorExtractor: AnyRef = codec.treeToValue(node, extractorClass).asInstanceOf[AnyRef]
 
-    anchorExtractor match {
-      case extractor: AnchorExtractorBase[_] =>
-          val extractorNode = node.get("extractor")
-          if (extractorNode != null && extractorNode.get("params") != null) {
-            // init the param into the extractor
-            val config = ConfigFactory.parseString(extractorNode.get("params").toString)
-            extractor.init(config)
+    val extractorNode = node.get("extractor")
+    if (extractorNode != null && extractorNode.get("params") != null) {
+      // init the param into the extractor
+      val config = ConfigFactory.parseString(extractorNode.get("params").toString)
+      anchorExtractor match {
+        case aebExtractor: AnchorExtractorBase[_] =>
+          aebExtractor.init(config)
+        case otherExtractor =>
+          FeathrUdfPluginContext.getRegisteredUdfAdaptor(extractorClass) match {
+            case Some(adaptor: SimpleAnchorExtractorSparkAdaptor) =>
+              adaptor.adaptUdf(otherExtractor).init(config)
```

Comment:
The adapted anchorExtractor is not used in the rest of code below. I believe we should update the anchorExtractor with the adapted anchorExtractor.

Reply:
Oops. I'll post a fix. I'm not going to have time to test these cases before I go on leave. Please make sure they're covered in your upcoming changes.